operator/watchers/cilium_podippool.go

```diff
@@ -19,6 +28,107 @@ type PooledAllocatorProvider interface {
+)
+
+// parsePoolSpec parses a pool spec string in the form
+// "ipv4-cidrs:172.16.0.0/16,172.17.0.0/16;ipv4-mask-size:24".
+func parsePoolSpec(poolString string) (cilium_v2alpha1.IPPoolSpec, error) {
+	fields := strings.Split(strings.ReplaceAll(poolString, " ", ""), ";")
+
+	var ipv4CIDRs, ipv6CIDRs []cilium_v2alpha1.PoolCIDR
+	var ipv4MaskSize, ipv6MaskSize uint64
+
+	for _, field := range fields {
+		key, value, ok := strings.Cut(field, ":")
+		if !ok {
+			return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid number of key delimiters in pool spec %s", poolString)
+		}
+		switch key {
+		case poolKeyIPv4CIDRs:
+			for _, cidr := range strings.Split(value, ",") {
+				_, err := netip.ParsePrefix(cidr)
+				if err != nil {
+					return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid value for key %q: %w", poolKeyIPv4CIDRs, err)
+				}
+				ipv4CIDRs = append(ipv4CIDRs, cilium_v2alpha1.PoolCIDR(cidr))
+			}
+		case poolKeyIPv4MaskSize:
+			mask, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid value for key %q: %w", poolKeyIPv4MaskSize, err)
+			}
+			ipv4MaskSize = mask
+		case poolKeyIPv6CIDRs:
+			for _, cidr := range strings.Split(value, ",") {
+				_, err := netip.ParsePrefix(cidr)
+				if err != nil {
+					return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid value for key %q: %w", poolKeyIPv6CIDRs, err)
+				}
+				ipv6CIDRs = append(ipv6CIDRs, cilium_v2alpha1.PoolCIDR(cidr))
+			}
+		case poolKeyIPv6MaskSize:
+			mask, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid value for key %q: %w", poolKeyIPv6MaskSize, err)
+			}
+			ipv6MaskSize = mask
+		}
+	}
+
+	pool := cilium_v2alpha1.IPPoolSpec{}
+	if len(ipv4CIDRs) > 0 {
+		pool.IPv4 = &cilium_v2alpha1.IPv4PoolSpec{
+			CIDRs:    ipv4CIDRs,
+			MaskSize: ipv4MaskSize,
+		}
+	}
+	if len(ipv6CIDRs) > 0 {
+		pool.IPv6 = &cilium_v2alpha1.IPv6PoolSpec{
+			CIDRs:    ipv6CIDRs,
+			MaskSize: ipv6MaskSize,
+		}
+	}
+
+	return pool, nil
+}
+
+func multiPoolAutoCreatePools(ctx context.Context, clientset client.Clientset, poolMap map[string]string) {
+	for poolName, poolSpecStr := range poolMap {
+		poolSpec, err := parsePoolSpec(poolSpecStr)
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				"poolName": poolName,
+				"poolSpec": poolSpecStr,
+			}).Fatalf("Failed to parse IP pool spec in %q flag", operatorOption.IPAMAutoCreateCiliumPodIPPools)
+		}
+
+		pool := &cilium_v2alpha1.CiliumPodIPPool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: poolName,
+			},
+			Spec: poolSpec,
+		}
+
+		_, err = clientset.CiliumV2alpha1().CiliumPodIPPools().Create(ctx, pool, metav1.CreateOptions{})
+		if err != nil {
+			if k8sErrors.IsAlreadyExists(err) {
+				// Nothing to do, we will not try to update an existing resource
+				log.WithField("poolName", poolName).Debug("CiliumPodIPPool resource already exists")
```

Comment:
I wonder whether this should be `Info` or even `Warning` such that it is more visible to users?

Reply:
I don't think it should be a warning. This is expected to happen if the operator restarts. Even a `info` is a bit noisy, which is why I made it a debug log statement. But I'll change it to Info again, to avoid any confusion as to why objects are not touched once they have been created.

Reply:
Good point. I didn't consider that this will be hit frequently during restart. Changing to Info or leaving as Debug both SHTM.